android/app/src/main/java/updated/mysterium/vpn/model/manual/connect/ProposalModel.kt

```diff
@@ -0,0 +1,82 @@
+package updated.mysterium.vpn.model.manual.connect
+
+import android.graphics.Bitmap
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+import network.mysterium.proposal.NodeType
+import network.mysterium.proposal.QualityLevel
+import network.mysterium.proposal.ServiceType
+import network.mysterium.service.core.ProposalPaymentMethod
+import network.mysterium.service.core.ProposalPaymentMoney
+import network.mysterium.service.core.ProposalPaymentRate
+import network.mysterium.ui.Countries
+import updated.mysterium.vpn.database.entity.NodeEntity
+import java.util.*
+
+@Parcelize
+data class ProposalModel(
+    val id: String,
+    val providerID: String,
+    val serviceType: ServiceType,
+    val countryCode: String,
+    val nodeType: NodeType,
+    val monitoringFailed: Boolean,
+    val payment: ProposalPaymentMethod,
+    val countryFlagImage: Bitmap?,
+    val qualityLevel: QualityLevel,
+    val countryName: String
+): Parcelable {
+
+    var priceLevel = PriceLevel.MEDIUM
+        private set
+
+    fun calculatePriceLevel(
+        minPrice: Double,
+        firstPriceBorder: Double,
+        secondPriceBorder: Double
+    ) {
+        priceLevel = when (payment.rate.perBytes) {
+            0.0 -> {
+                PriceLevel.FREE
+            }
+            in minPrice..firstPriceBorder -> {
+                PriceLevel.LOW
+            }
+            in firstPriceBorder..secondPriceBorder -> {
+                PriceLevel.MEDIUM
+            }
+            else -> {
+                PriceLevel.HIGH
+            }
+        }
+    }
+
+    companion object {
+
+        fun createProposalFromNode(nodeEntity: NodeEntity): ProposalModel {
```

Comment:
Instead of creating static methods create custom constructors for `ProposalModel`. So it will look like this:
```
constructor(nodeEntity: NodeEntity) : this(
    nodeEntity.countryCode.toLowerCase(Locale.ROOT),
    ProposalPaymentMethod(
                type = nodeEntity.paymentType,
                price = ProposalPaymentMoney(nodeEntity.paymentAmount, nodeEntity.currency),
                rate = ProposalPaymentRate(nodeEntity.pricePerSecond, nodeEntity.pricePerByte)
    ),
    // and rest of code
)
```

Reply:
Fix